src/Symfony/Component/AssetMapper/Compressor/ZopfliCompressor.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\Compressor;
+
+use Symfony\Component\Process\Process;
+
+/**
+ * Compresses a file using zopfli.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+final class ZopfliCompressor implements SupportedCompressorInterface
+{
+    use CompressorTrait;
+
+    private const WRAPPER = ''; // not supported yet https://github.com/kjdev/php-ext-zopfli/issues/23
+    private const COMMAND = 'zopfli';
+    private const PHP_EXTENSION = '';
+    private const FILE_EXTENSION = 'gz';
+
+    public function __construct(
+        ?string $executable = null,
+    ) {
+        $this->executable = $executable;
+    }
+
+    private function compressWithBinary(string $path): void
+    {
+        (new Process([$this->executable, '--', $path]))->mustRun();
+    }
+
+    private function createStreamContext()
```

Comment:
@dunglas this method with no return type and no phpdoc `@return` makes our CI fail for the check about our migration to native return types.

Reply:
The type is documented in the PHPDoc of the abstract function in the trait, but I can add it here too if needed.